src/core_plugins/metrics/common/calculate_label.js

```diff
@@ -28,10 +28,19 @@ export default function calculateLabel(metric, metrics) {
     return `${lookup[metric.type]} (${metric.value}) of ${metric.field}`;
   }
 
-  if (_.includes(paths, metric.type)) {
-    const targetMetric = _.find(metrics, { id: metric.field });
+
+  if (includes(paths, metric.type)) {
+    let additionalLabel = '';
+    const targetMetric = metrics.find(m => startsWith(metric.field, m.id));
     const targetLabel = calculateLabel(targetMetric, metrics);
-    return `${lookup[metric.type]} of ${targetLabel}`;
+    if (targetMetric.type === 'percentile') {
```

Comment:
This doesn't work if you add multiple derivatives, because there's a chance `targetMetric` is undefined. We're tracking the `field` on the metric object, but when adding a 2nd derivative configuration, that second one hasn't necessarily been fully configured yet.

Reply:
I added a check for `targetMetric` which fixes the issue